optuna/artifacts/_boto3.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import io
+import shutil
+from typing import TYPE_CHECKING
+
+import boto3
+from botocore.exceptions import ClientError
+
+from optuna.artifacts.exceptions import ArtifactNotFound
+
+
+if TYPE_CHECKING:
+    from typing import BinaryIO
+
+    from mypy_boto3_s3 import S3Client
+
+
+class Boto3ArtifactStore:
+    """An artifact backend for Boto3."""
```

Comment:
How about adding the explanations for arguments?

Reply:
Thanks for the comments. I added more explanations.